rule.php

```diff
@@ -227,32 +234,56 @@ public static function get_settings_sql($quizid) : array {
     }
 
 
-
     /**
      * Sets up the attempt (review or summary) page with any special extra
      * properties required by this rule.
      *
      * @param moodle_page $page the page object to initialise.
      */
-    public function setup_attempt_page($page) {
+    public function setup_attempt_page($page)
+    {
         $page->set_title($this->quizobj->get_course()->shortname . ': ' . $page->title);
         $page->set_heading($page->title);
-        $proctortype=$this->quizobj->get_quiz()->proctortype;
+        $proctortype = $this->quizobj->get_quiz()->proctortype;
         $local_proview_enabled = get_config('local_proview', 'enabled');
         $quizaccess_proctor_setting_enabled = get_config('quizaccess_proctor', 'enableproctor');
-        if ( $local_proview_enabled && $quizaccess_proctor_setting_enabled && $proctortype !== 'noproctor'){
+        if ($local_proview_enabled && $quizaccess_proctor_setting_enabled && $proctortype !== 'noproctor') {
             $page->set_pagelayout('secure');
             $page->set_popup_notification_allowed(false); // Prevent message notifications.
         }
+
+        if ($this->is_tsb_required() && !strpos($_SERVER ['HTTP_USER_AGENT'], "Proview-SB")) {
+            $this->redirect_to_tsb_link();
+        }
     }
 
     /**
      * This is called when the current attempt at the quiz is finished.
      */
-    public function current_attempt_finished() {
+    public function current_attempt_finished()
+    {
         $this->accessmanager->clear_session_access();
     }
 
+    public function is_tsb_required()
+    {
+        $quizaccess_proctor_setting_enabled = get_config('quizaccess_proctor', 'enableproctor');
+        $tsbenabled = $this->quizobj->get_quiz()->tsbenabled;
+        return ($quizaccess_proctor_setting_enabled && $tsbenabled);
```

Comment:
@rohansharmasitoula Also add a check for `$proctortype !== 'noproctor'`

Reply:
Added @devang1281